server/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java

```diff
@@ -3927,9 +3926,7 @@ public void testScheduledRefresh() throws Exception {
         assertTrue(primary.searchIdleTime() >= TimeValue.ZERO.millis());
         primary.flushOnIdle(0);
         logger.info("--> scheduledRefresh(future5)");
-        PlainActionFuture<Boolean> future5 = new PlainActionFuture<>();
-        primary.scheduledRefresh(future5);
-        assertTrue(future5.actionGet()); // make sure we refresh once the shard is inactive
+        assertTrue(primary.scheduledRefresh(ActionListener.noop())); // make sure we refresh once the shard is inactive
```

Comment:
I preferred the assertBusy solution you did initially, which confirms that the refresh is truly executed.

The current code asserts the refresh is sent, but the refresh happens asynchronously and races with the subsequent searcher (so in some cases it might see 2 docs and in others 3 docs).